### ISSUE TITLE
Running in gossip mode on mainnet results in failed to verify protocol message signature InvalidPullRequest

### ISSUE THREAD

**Issue by clarkezone** [Syndica/sig#61]
### Description

Building from tip for tree and running in gossipmode against mainnet results in:

`time=2023-12-21T17:11:34Z level=debug  failed to verify protocol message signature InvalidPullResponse `

Not sure if this is expected based on current state of codebase or not hence filing issue

### How to Reproduce the Bug

1. Clone 949eb15dbecc1cbd091ccf8344aa5c65b16278a8
2. Follow instructions from new (much better!) readme.md to run against mainnet
3. Result: above error message.

**Comment by dnut** [comment]
Did you make any changes to the code before running this?

With spy_node = false, I see a similar log message that I'd like to investigate:

```
time=2023-12-22T21:23:15Z level=debug  gossip: packet_verify: failed to verify signature
```

**Comment by clarkezone** [comment]
No changes and I didn't change any flags

**Comment by dnut** [comment]
So far, I've narrowed this down to be an issue with ContactInfo pull responses.

All other pull responses are working fine. Every ContactInfo pull response received from either mainnet or devnet fails signature validation. Hundreds of messages are succeeding the signature validation for every one failed ContactInfo.

Like the rust implementation, Sig expects the ContactInfo to be signed by the private key corresponding to the public key contained within the ContactInfo. But the signature included alongside the ContactInfo is failing that validation. We have some tests on serialization/deserialization, plus I just added one more, and I haven't found a problem there. I'll have to keep digging to figure this out.
